scripts/globals/pasteventwatcher.lua

```diff
@@ -0,0 +1,60 @@
+-----------------------------------
+-- Past Event Watcher NPCs
+-----------------------------------
+
+-----------------------------------
+
+dsp = dsp or {}
+dsp.pasteventwatcher = dsp.pasteventwatcher or {}
+
+dsp.pasteventwatcher.goblinFootprint = function(options)
+    -- Defaults options to 0 if not all of them have been passed.
+    for i=1,31, 1 do
+        if not options[i] then
+            options[i] = 0
+        end
+    end
+
+    -- 32 bits represents a cutscene being watched or not
+    -- The first bit defaults to 0 to show the "Another time maybe" choice to let user get out from the dialog
+    -- The first scene starts at bit2, and it has to be 0 if it becomes part of the list, otherwise it should be a 1.
+    -- If you need to know the correct order of the scenes use !cs <scene number> for the gobbie to list them all
+    local bit32 = (1-options[31])*2147483648
+    local bit31 = (1-options[30])*1073741824
+    local bit30 = (1-options[29])*536870912
+    local bit29 = (1-options[28])*268435456
+    local bit28 = (1-options[27])*134217728
+    local bit27 = (1-options[26])*67108864
+    local bit26 = (1-options[25])*33554432
+    local bit25 = (1-options[24])*16777216
+    local bit24 = (1-options[23])*8388608
+    local bit23 = (1-options[22])*4194304
+    local bit22 = (1-options[21])*2097152
+    local bit21 = (1-options[20])*1048576
+    local bit20 = (1-options[19])*524288
+    local bit19 = (1-options[18])*262144
+    local bit18 = (1-options[17])*131072
+    local bit17 = (1-options[16])*65536
+    local bit16 = (1-options[15])*32768
+    local bit15 = (1-options[14])*16384
+    local bit14 = (1-options[13])*8192
+    local bit13 = (1-options[12])*4096
+    local bit12 = (1-options[11])*2048
+    local bit11 = (1-options[10])*1024
+    local bit10 = (1-options[9])*512
+    local bit9  = (1-options[8])*256
+    local bit8  = (1-options[7])*128
+    local bit7  = (1-options[6])*64
+    local bit6  = (1-options[5])*32
+    local bit5  = (1-options[4])*16
+    local bit4  = (1-options[3])*8
+    local bit3  = (1-options[2])*4
+    local bit2  = (1-options[1])*2
+    local bit1  = 1-1*1
```

Comment:
https://www.youtube.com/watch?v=U1UtRnGn5hc

Reply:
All of this should be using bitwise operations. They are infinitely faster. Brush up on them and how to do them in lua.

Reply:
How bout an example of what you'd like? I'm sure if he knew what was expected it would have already been done that way. thanks to @ibm2431 for being very useful to people who just want to help and contribute

Reply:
>  I'm sure if he knew what was expected it would have already been done that way

Common sense is a good place to start, if it needs to be zero, use a zero.

Reply:
So what I am taking from all of this is: What this overly bloated and unnecessarily demeaning review should have said is:
```
use 0 here:
local bit1  = 1-1*1
```

Reply:
Take your white knight shit elsewhere.

Reply:
Gladly! Kindly point me in the direction of somewhere else where they demean, bully, and ridicule for fun... :thinking:

Reply:
@teschnei you condone all of this?

Reply:
a DSP OP bans me from discord for this thread

Reply:
you deleted everything except the video mocking this poor guys mistake? @teschnei

Reply:
for the bitwise functions, we automatically require the "bit" library in every Lua script, so you can use `bit.and` etc. without needing to require it (this one, I think: http://bitop.luajit.org/)